### PR TITLE
Added method for getting every unique permuatation

### PR DESCRIPTION
**Pull request by ashotjanibekyan**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by ashotjanibekyan** [comment]
Sorry for having a lot of commits, I'm new to pull requests.

Should I create a new branch for every new algorithm in the future?

File: Algorithms.Tests/Strings/PermutationTests.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using Algorithms.Numeric;
+using Algorithms.Strings;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Strings
+{
+    public class AnagramTests
+    {
+
```
Comment (siriak) [inline Algorithms.Tests/Strings/PermutationTests.cs:13]:
```suggestion
```

File: Algorithms.Tests/Strings/PermutationTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        [TestCase("abcd")]
+        [TestCase("aabcd")]
+        [TestCase("aabbbcd")]
+        [TestCase("aabbbccccd")]
+        public void Test_GetEveryUniquePermutation(string word)
+        {
+            var anagrams = Permutation.GetEveryUniquePermutation(word);
+
+            // make sure we have the right number of anagrams
+            var occDic = new Dictionary<char, int>();
```
Comment (siriak) [inline Algorithms.Tests/Strings/PermutationTests.cs:25]: Please provide a better name, it's not clear what occ is and Dic is redundant if it means dictionary.

**Review by siriak — CHANGES_REQUESTED** [review]

**Comment by siriak** [comment]
No problem in having many commits, they'll be squashed in one anyway :)

**Comment by ashotjanibekyan** [comment]
@siriak, thanks for the feedback. I added comments to clarify my intent. Don't remember if the formula had a name (discrete math was years ago), otherwise, I would have added some references too. Is that necessary?

**Review by siriak — APPROVED** [review]
Great explanation, thanks a lot!

**Pull request merged**